omod/src/test/java/org/openmrs/module/idgen/web/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,137 @@
+    assertEquals(autoGenerationOption.getSource(), PropertyUtils.getProperty(result, "source"));
+    assertEquals(autoGenerationOption.isAutomaticGenerationEnabled(), PropertyUtils.getProperty(result, "isAutomaticGenerationEnabled"));
+    assertEquals(autoGenerationOption.isManualEntryEnabled(), PropertyUtils.getProperty(result, "isManualEntryEnabled"));
+  }
+
+  @Test
+  public void shouldGetAnAutoGenerationOptionByPatientIdentifierType() throws Exception {
+    MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + ptid);
+    SimpleObject result = deserialize(handle(req));
+    
+    AutoGenerationOption autoGenerationOption = service.getAutoGenerationOption(ptid);
+    assertEquals(autoGenerationOption.getId(), PropertyUtils.getProperty(result, "id"));
+    assertEquals(autoGenerationOption.getIdentifierType(), PropertyUtils.getProperty(result, "identifierType"));
+    assertEquals(autoGenerationOption.getLocation(), PropertyUtils.getProperty(result, "location"));
+    assertEquals(autoGenerationOption.getSource(), PropertyUtils.getProperty(result, "source"));
+    assertEquals(autoGenerationOption.isAutomaticGenerationEnabled(), PropertyUtils.getProperty(result, "isAutomaticGenerationEnabled"));
+    assertEquals(autoGenerationOption.isManualEntryEnabled(), PropertyUtils.getProperty(result, "isManualEntryEnabled"));
+  }
+
+  @Test
+  public void shouldCreateNewAutoGenerationOption() throws Exception {
+    long originalCount = getAllCount();
+
+    SimpleObject autoGenerationOption = new SimpleObject();
+    autoGenerationOption.add("identifierType", "idgen");
+    autoGenerationOption.add("location", "National Hospital");
+    autoGenerationOption.add("source", "New source");
+    autoGenerationOption.add("isAutomaticGenerationEnabled", true);
+    autoGenerationOption.add("isManualEntryEnabled", true);
+
+    String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+
+    MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+    req.setContent(json.getBytes());
+    
+    SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "id"));
+		assertEquals(originalCount + 1, getAllCount());
+  }
+
+  @Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+    final String newLocation = "Updated Location";
+    final String newSource = "Updated Source";
+		SimpleObject autoGenerationOption = new SimpleObject();
+    autoGenerationOption.add("location", newLocation);
+    autoGenerationOption.add("source", newSource);
+		
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+    assertEquals(newLocation, service.getAutoGenerationOption(Integer.parseInt(getUuid())).getLocation());
```

Comment:
Same applies here as i have commented before. In summary, if you assert something after making a REST call that changes data, ensure that you have corresponding assertions before the REST call which prove that the opposite was true. Does this make sense to you?